managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java

```diff
@@ -126,10 +127,18 @@ private ManagedLedgerFactoryImpl(BookKeeper bookKeeper, boolean isBookkeeperMana
                 .numThreads(config.getNumManagedLedgerSchedulerThreads())
                 .name("bookkeeper-ml-scheduler")
                 .build();
-        orderedExecutor = OrderedExecutor.newBuilder()
-                .numThreads(config.getNumManagedLedgerWorkerThreads())
-                .name("bookkeeper-ml-workers")
-                .build();
+
+        if (bookKeeper.getMainWorkerPool() != null) {
```

Comment:
This is a public interface, but not a good one. And ownership is all messed up.

Does this even give a real performance benefit? The managed ledger selects the thread based on ml name, the ledger by the ledger id. There's no guarantee these will be the same. A better approach would be to allow the managed ledger to pass in its executor to the ledger on creation.